packages/core-data/src/actions.js

```diff
@@ -220,7 +225,11 @@ export function* saveEntityRecord(
 	kind,
 	name,
 	record,
-	{ isAutosave = false } = { isAutosave: false }
+	{
+		isAutosave = false,
+		getSuccessNoticeActionArgs,
+		getFailureNoticeActionArgs,
```

Comment:
There is no simple way to guarantee that they are called after the API request succeeds or fails or to access the necessary parameters internal to core data.

Reply:
I feel we can return a value maybe from this generator and retrieve it in the caller or better persist those parameters in the state (we do have isSaving, we can have more metadata about the save request)

Reply:
>I feel we can return a value maybe from this generator and retrieve it in the caller

Backwards compatibility requires that this returns the updated record. We would have to return the error here instead when it fails to infer that there is one? Should it be in an object? What if the entity has an `.error` property? It just really complicates things and makes life way harder for the user for a little semantic improvement.

Reply:
Right! that makes sense.

Looking at the reducer, we already have a `saving` reducer to track that contains the status, and the error. Isn't that enough?

Reply:
That's only there for legacy reasons, you would be imposing that requirement on any caller that wants to fire notices.

Reply:
> you would be imposing that requirement on any caller that wants to fire notices.

It doesn't sound like a big deal to me, if you just triggered a save action, you know you can call a selector to get its status. Am I missing something, it feels better than having a weird public API for the action?

Reply:
The caller isn't necessarily the editor store. A plugin might save an entity in a myriad of different ways outside of a block list or even an editor. You would be forcing the caller to be stateful and keep track of the entity before saving, and its saving/error state.

The ideal approach here would be for an object with a `result` or `data`, and `error` properties to be returned, but doing that would break backwards compatibility.

Reply:
> You would be forcing the caller to be stateful and keep track of the entity before saving, and its saving/error state.

I was suggesting we store the state in the "core" data module and not the caller. The caller would just do something like:

```
yield saveEntityRecord()
const saveState = yield select( 'core' ).getEntityRecordSaveState();
```

Reply:
https://github.com/WordPress/gutenberg/pull/16932/commits/4c998a2fb3ac604dd55057d387f0edf50199c031

Reply:
Thanks for the update, that looks better to me 👍